homeassistant/components/remote/harmony.py

```diff
@@ -0,0 +1,176 @@
+        return self._state
+
+    @property
+    def token(self):
+        """Return the token of the Harmony device."""
+        return self._token
+
+    @property
+    def config(self):
+        """Return the configuration of the Harmony device."""
+        return self._config
+
+    @property
+    def state_attributes(self):
+        """Overwrite inherited attributes."""
+        return {'current_activity': self._current_activity,
+                'authtoken': self._token}
+
+    def is_on(self):
+        """Return False if PowerOff is the current activity, otherwise true."""
+        return self.update()
+
+    def update(self):
+        """Return current activity."""
+        import pyharmony
+        _LOGGER.debug('polling hub at for current activity')
+        state = pyharmony.ha_get_current_activity(self._token,
+                                                  self._config,
+                                                  self._ip,
+                                                  self._port)
+        _LOGGER.debug('current activity reported as: ' + state)
+        self._current_activity = state
+        if state != 'PowerOff':
+            self._state = STATE_ON
+        else:
+            self._state = STATE_OFF
+
+    def turn_on(self, **kwargs):
+        """Start an activity from the Harmony device."""
+        import pyharmony
+        if kwargs[ATTR_ACTIVITY]:
+            activity = kwargs[ATTR_ACTIVITY]
+        else:
+            activity = self._default_activity
+
+        if activity != 'Not_Set':
+            pyharmony.ha_start_activity(self._token,
+                                        self._ip,
+                                        self._port,
+                                        self._config,
+                                        activity)
+            self._state = STATE_ON
+        else:
+            _LOGGER.error('No activity specified with turn_on service')
+
+    def turn_off(self):
+        """Start the PowerOff activity."""
+        import pyharmony
+        pyharmony.ha_power_off(self._token, self._ip, self._port)
+
+    def send_command(self, **kwargs):
+        """Send a command to one device."""
+        import pyharmony
+        pyharmony.ha_send_command(self._token, self._ip,
+                                  self._port, kwargs[ATTR_DEVICE],
+                                  kwargs[ATTR_COMMAND])
+
+    def sync(self):
+        """Sync the Harmony device with the web service."""
+        import pyharmony
+        _LOGGER.debug('syncing hub with Harmony servers')
+        pyharmony.ha_sync(self._token, self._ip, self._port)
+        self._config = pyharmony.ha_get_config(self._token,
+                                               self._ip,
+                                               self._port)
+        _LOGGER.debug('writing hub config to file: ' + self._config_path)
+        pyharmony.ha_get_config_file(self._config, self._config_path)

```

Comment:
The function name indicates it is getting something but the log message is indicating it is writing something ?